src/types/resolveConstantValue.ts

```diff
@@ -71,13 +71,16 @@ function reduceInt(ast: ASTExpression): bigint {
         return reduceIntImpl(ast);
     } catch (error) {
         if (error instanceof RangeError) {
-            throwError(
-                "Cannot evaluate constant expression due to integer overflow",
-                ast.ref,
-            );
-        } else {
-            throw error;
+            if (error.message === "Division by zero") {
+                throwError("Cannot divide by zero", ast.ref);
+            } else if (error.message === "Maximum BigInt size exceeded") {
```

Comment:
What about 257-bit integers? Or integers with storage modifiers like `Int as uint32`?

Reply:
257-bit integers should work fine, also I didn't add the `Maximum BigInt size exceeded`, it was added before in #200, so why would it stop working correctly?

Reply:
No, I meant, BigInts in TypeScript are not 257-bit integers, this is why "Maximum BigInt size exceeded" looks a bit weird here

Reply:
But it's the error that TypeScript throws, and when we catch it in Tact, an "Cannot evaluate constant expression due to integer overflow" compilation error is thrown instead.